### ISSUE TITLE
Document transient publishes once more client libraries implement it

### ISSUE THREAD

**Issue by SimonWoolf** [ably/docs#357]
currently only implemented in ably-js

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/EDU-1716) by [Unito](https://www.unito.io)

**Comment by mattheworiordan** [comment]
This is perhaps not something we should do until the next version of our spec is released.  Arguably, this is an experimental feature until officially supported.

**Comment by mattheworiordan** [comment]
Related to https://github.com/ably/docs/issues/319 in regards to how we recommend customers publish large volumes of messages.  Closing that issue in favour of this.

**Comment by mattheworiordan** [comment]
@QuintinWillison I assume we do support transient publishing now in all libs and can close this (if documented)?

**Comment by QuintinWillison** [comment]
It was documented in 27157ba673aecbf56ee808cbc1b94e38380864a4 but without Java, Objective-C or C# examples. We [claim support](https://www.ably.io/download/sdk-feature-support-matrix) in both Java and .NET, however I can only find one instance of the word 'transient' [in the spec](https://docs.ably.io/client-lib-development-guide/features/#TR3e) so I'm not entirely clear what I'm looking for.

Java, for example, doesn't have anywhere near the number of members I would expect in its [`ProtocolMessage.Flag` enum](https://github.com/ably/ably-java/blob/develop/lib/src/main/java/io/ably/lib/types/ProtocolMessage.java#L53).

Before exploring deeper... @mattheworiordan, can you give me any more pointers as to what I should be looking for here? Where is this feature defined?

**Comment by QuintinWillison** [comment]
ok, as I clunk through my GitHub notifications I see you've indirectly given me the answer with https://github.com/ably/docs/pull/323 ... "published immediately" and "implicit attach" from [RTL6](https://docs.ably.io/client-lib-development-guide/features/#RTL6) sub-items.

**Comment by paddybyers** [comment]
The spec change can be found via https://github.com/ably/docs/pulls?utf8=%E2%9C%93&q=is%3Apr+transient+publishing

The java implementation, for example, can be found via https://github.com/ably/ably-java/pulls?utf8=%E2%9C%93&q=is%3Apr+transient+publishing

I thought that we had specified that as a required feature for any library advertising itself as 1.1, so it should be implemented in most or all libraries. Having said that, I don't remember seeing it in PHP or Python.

**Comment by SimonWoolf** [comment]
> Having said that, I don't remember seeing it in PHP or Python.

It's a realtime client lib feature, php and python are currently rest-only

**Comment by paddybyers** [comment]
> It's a realtime client lib feature, php and python are currently rest-only

Yes of course, I was thinking of idempotent publishing for some reason.
